internal/statemachine/helper.go

```diff
@@ -999,6 +999,20 @@ func (stateMachine *StateMachine) updateGrub(rootfsVolName string, rootfsPartNum
 	// make sure to unmount the disk too
 	umounts = append(umounts, exec.Command("umount", mountDir))
 
+	// divert GRUB's os-prober as we don't want to scan for other OSes on
+	// the build system
+	updateGrubCmds = append(updateGrubCmds,
```

Comment:
Could we also have an unit test for this? Not sure yet what test case would be ideal here - the bare minimum would be maybe to at least inspect the commands that are being executed, but that feels a bit 'lacking'. Since best would be if we could actually check if `os-prober` is running or not, or something. Could you maybe look into that? If it's too difficult, we can look into it later.